File: app/src/main/java/com/wang/example/msg/bean/TextBean.java
package com.wang.example.msg.bean;

import androidx.annotation.NonNull;

import com.alibaba.fastjson.JSON;
import com.wang.container.adapter.IContainerItemAdapter;
import com.wang.example.msg.adapter.TextAdapter;

public class TextBean extends BaseMsgBean {
    public TextDataEntity textInfo;

    protected TextBean(BaseMsgBean oldBean) {
        super(oldBean);
        textInfo = JSON.parseObject(oldBean.data, TextDataEntity.class);
    }

    public static class TextDataEntity {
        public String text;
    }

    @NonNull
    @Override
    public Class<? extends IContainerItemAdapter> getBindAdapterClass() {
        return TextAdapter.class;
    }
}

File: app/src/main/java/com/wang/example/msg/bean/TextBean.kt
package com.wang.example.msg.bean

import com.google.gson.Gson
import com.wang.example.msg.adapter.TextAdapter

class TextBean(oldBean: BaseMsgBean) : BaseMsgBean(oldBean) {
    var textInfo: TextDataEntity

    class TextDataEntity {
        var text: String? = null
    }

    override fun getBindAdapterClass() = TextAdapter::class.java

    init {
        textInfo = Gson().fromJson(oldBean.data, TextDataEntity::class.java)
    }
}